File: frontend/src/api/company.ts
import { apiClient } from './client';

export interface Company {
  id: string;
  name: string;
  slug: string;
  website?: string;
  verification_doc_url?: string;
  employee_proof_doc_url?: string;
  is_verified: boolean;
  created_by_user_id?: string;
  median_response_hours?: number;
  ghosting_rate?: number;
  interview_to_offer_rate?: number;
  total_tracked_applications: number;
  created_at: string;
}

export const getCompanies = async (): Promise<Company[]> => {
  const response = await apiClient.get('/companies/');
  return response.data;
};

export const createCompany = async (data: { name: string; website?: string; verification_doc_url?: string; employee_proof_doc_url?: string }): Promise<Company> => {
  const response = await apiClient.post('/companies/', data);
  return response.data;
};

File: backend/app/models/company.py
from sqlmodel import SQLModel, Field
from uuid import UUID, uuid4
from datetime import datetime
from typing import Optional

class Company(SQLModel, table=True):
    id: UUID = Field(default_factory=uuid4, primary_key=True)
    name: str = Field(index=True)
    slug: str = Field(unique=True, index=True)
    website: Optional[str] = None
    verification_doc_url: Optional[str] = None
    employee_proof_doc_url: Optional[str] = None
    is_verified: bool = Field(default=False)
    created_by_user_id: Optional[UUID] = Field(default=None, foreign_key="user.id", index=True)
    median_response_hours: Optional[float] = None
    ghosting_rate: Optional[float] = None
    interview_to_offer_rate: Optional[float] = None
    total_tracked_applications: int = Field(default=0)
    score_last_computed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
